projects/sunbird-player-sdk/src/lib/player-utils/service/error/error.service.ts

```diff
@@ -0,0 +1,34 @@
+import { EventEmitter, Injectable } from '@angular/core';
+import { errorCode, errorMessage } from './../../enums/exceptionLogs';
+@Injectable({
+  providedIn: 'root'
+})
+export class ErrorService {
+  playerContentCompatibiltyLevel = 4;
+  getInternetConnectivityError = new EventEmitter<any>();
+
+  constructor() {
+    this.initInternetConnectivityError();
+  }
+
+  checkContentCompatibility(currentCompatibilityLevel: number) {
+    if (currentCompatibilityLevel > this.playerContentCompatibiltyLevel) {
+      const compatibilityError = new Error();
+      compatibilityError.message = `Player supports ${this.playerContentCompatibiltyLevel}
+      but content compatibility is ${currentCompatibilityLevel}`;
+      compatibilityError.name = 'contentCompatibily';
+      return { error: compatibilityError, isCompitable: false };
+    } else {
+      return { error: null, isCompitable: true };
+    }
+  }
+
+  initInternetConnectivityError() {
+    window.addEventListener('offline', (e) => {
```

Comment:
Angular service can have the onDestory lifecycle hook so move this listener method to separate private method and remove the listener in onDestory call of service. 

that means like this 

ngOnDestory() {
   window.removeEventListener('offline', <method>);
}